src/transactions/instances/transactionPayload.ts

```diff
@@ -3,16 +3,47 @@
 
 /* eslint-disable @typescript-eslint/naming-convention */
 
-import { Serializer, Deserializer, Serializable } from "../../bcs";
+import { Deserializer } from "../../bcs/deserializer";
+import { Serializable, Serializer } from "../../bcs/serializer";
 import { AccountAddress } from "../../core";
 import { Identifier } from "./identifier";
-import { ScriptTransactionArgument } from "./scriptTransactionArguments";
 import { ModuleId } from "./moduleId";
-import { TransactionPayloadVariants } from "../../types";
+import { ScriptTransactionArgumentVariants, TransactionPayloadVariants } from "../../types";
 import { TypeTag } from "../typeTag/typeTag";
-import { U8 } from "../../bcs/serializable/move-primitives";
+import type { EntryFunctionArgument, ScriptFunctionArgument, TransactionArgument } from "./transactionArgument";
+import { EntryFunctionBytes } from "../../bcs/serializable/entry-function-bytes";
+import { Bool, U128, U16, U256, U32, U64, U8 } from "../../bcs/serializable/move-primitives";
 import { MoveVector } from "../../bcs/serializable/move-structs";
-import { FixedBytes } from "../../bcs/serializable/fixed-bytes";
+
+/**
+ * Deserialize a Script Transaction Argument
+ */
+export function deserializeFromScriptArgument(deserializer: Deserializer): TransactionArgument {
```

Comment:
why place it in this file?

Reply:
I wasn't sure where to put it in order to avoid circular deps, but I figured it's *only* used for script payloads so just put it here